### PR TITLE
doc: update erasure coding document with overhead formula

### PR DESCRIPTION
**Pull request by vumrao**
doc: update erasure coding document with overhead formula
Signed-off-by: Vikhyat Umrao <vumrao@redhat.com>

### CODE REVIEW

File: doc/rados/operations/erasure-code.rst [hunk 1]
```diff
@@ -115,6 +115,20 @@ no two *chunks* are stored in the same rack.
 More information can be found in the `erasure code profiles
 <../erasure-code-profile>`_ documentation.
 
+Erasure code profiles overhead calculation
+------------------------------------------
```
Comment (neha-ojha) [inline doc/rados/operations/erasure-code.rst]: https://docs.ceph.com/docs/master/rados/operations/erasure-code/#erasure-code-profiles section talks a bit about overhead, how about we use the same example and move this there?

File: doc/rados/operations/erasure-code.rst [hunk 2]
```diff
@@ -115,6 +115,20 @@ no two *chunks* are stored in the same rack.
...
+
+Formula for calculating the percentage overhead::
+
+    *K*/*K*+*M* * 100
```
Comment (ivancich) [inline doc/rados/operations/erasure-code.rst]: Due to order of operations, I'd like to see *K*/(*K* + *M*) .

File: doc/rados/operations/erasure-code.rst [hunk 3]
```diff
@@ -115,6 +115,20 @@ no two *chunks* are stored in the same rack.
...
+
+For example for profile (*K=8*) and (*M=3*) the percentage overhead is 72.73%.
+
+Formula for calculating the raw usage overhead::
+
+    *K*/*K*+*M* * (total raw usage)
```
Comment (ivancich) [inline doc/rados/operations/erasure-code.rst]: Here too -- due to order of operations, I'd like to see *K*/(*K* + *M*) .

**Comment by jdurgin** [comment]
@ivancich I agree with your analysis. The k/(k+m) formula is the reciprocal of 1 + overhead, which is the effective space available when multiplied by raw storage.

m/k as the plain overhead is correct.

**Comment by vumrao** [comment]
Thanks, Josh, Neha and Eric for the feedback. I have tried to cover both cases, please let me know your inputs.

File: doc/rados/operations/erasure-code.rst [hunk 4]
```diff
@@ -61,6 +61,31 @@ the following profile can be defined::
     $ rados --pool ecpool get NYAN -
     ABCDEFGHI
 
+Formula for calculating the raw size percentage overhead::
+
+    K/(K+M) * 100
+
+For example for profile (*K=3*) and (*M=2*) the percentage overhead is 60%.
+
+Formula for calculating the raw size overhead::
```
Comment (ivancich) [inline doc/rados/operations/erasure-code.rst]: Using the term "overhead" here is incorrect. As @jdurgin said, this ratio helps calculate the "effective space available" given the raw space.

File: doc/rados/operations/erasure-code.rst [hunk 5]
```diff
@@ -61,6 +61,31 @@ the following profile can be defined::
     $ rados --pool ecpool get NYAN -
     ABCDEFGHI
 
+Formula for calculating the raw size percentage overhead::
```
Comment (ivancich) [inline doc/rados/operations/erasure-code.rst]: Using the term "overhead" here is incorrect. As @jdurgin said, this ratio helps calculate the "effective space available", although when multiplying by 100, it gives the "percentage of effective space available".

File: doc/rados/operations/erasure-code.rst [hunk 6]
```diff
@@ -61,6 +61,31 @@ the following profile can be defined::
...
+
+    K/(K+M) * 100
+
+For example for profile (*K=3*) and (*M=2*) the percentage overhead is 60%.
+
+Formula for calculating the raw size overhead::
+
+    K/(K+M) * (total raw size)
+
+For example for profile (*K=3*) and (*M=2*) for total raw size 256T, the raw size overhead is 153.6T.
```
Comment (ivancich) [inline doc/rados/operations/erasure-code.rst]: "overhead" should be "effective space available"

File: doc/rados/operations/erasure-code.rst [hunk 7]
```diff
@@ -61,6 +61,31 @@ the following profile can be defined::
...
+
+Formula for calculating the plain used overhead::
```
Comment (ivancich) [inline doc/rados/operations/erasure-code.rst]: I think "plain used" is here to create a different term from the above. But I think you can get rid of "plain used" and just call this "overhead".

File: doc/rados/operations/erasure-code.rst [hunk 8]
```diff
@@ -61,6 +61,31 @@ the following profile can be defined::
...
+
+    M/K * (plain used size)
```
Comment (ivancich) [inline doc/rados/operations/erasure-code.rst]: I don't think "plain used" helps here. And a reader might ask "size of what?". I don't think "plain used" helps below, either.

**Comment by ivancich** [comment]
If I can quote @jdurgin above "The k/(k+m) formula is the **reciprocal of 1 + overhead**". Therefore "k/(k+m)" is not the "overhead".

**Comment by ivancich** [comment]
Let's unstale this. @vumrao -- do you plan on returning to this?

**Comment by vumrao** [comment]
Thanks, @ivancich yeah I will fix it. @neha-ojha and I were discussing what is the best way to simply document it. I will follow up with her and will send an updated doc patch.

**Pull request closed without merging**